File: Pregunta 1/fibonacci.hs
--Fibonacci Recursivo
fibo :: Int -> Int
fibo 0 = 0
fibo 1 = 1
fibo 2 = 1
fibo n = (fibo (n-1)) + (fibo (n-2))

fiboR = map fibo [0..10]

main :: IO()
main = do
  putStrLn "Fibonacci Recursivo"
  print(fiboR)
  --Fibonacci Estructural 
  putStrLn "Fibonacci Estructural"
  print $ map fiboN [0..10]

fiboN 0 = 0
fiboN 1 = 1
fiboN 2 = 1
fiboN n = iter 0 1 n where
    iter a b 2 = a + b
    iter a b n = iter b (a + b) (n - 1)  
  
  
  

File: Pregunta 1/fibonacci.scala
object fibonacci {
  def fibo(n:Int):Unit={
    if(n>0){
     while(i <= n)
      c = a + b;
      a=b
      b=c
      print(" "+ c )
      i=i+1
    }
  }
  var a = 0
  var b = 1
  var c = 0
  var i = 1 
  
  def fib(x:Int) = {
    var msg= "%d "
    msg.format(fibonacci(x))
  }
  
  def fibonacci(n: Int): Int = {
    if ( n == 0 ) 0
    else if ( n == 1 ) 1
    else fibonacci( n-1 ) + fibonacci( n-2 )

  }

  def main(args: Array[String]): Unit =
   println("Dame un valor n: ")
   val n =scala.io.StdIn.readInt()
   
   println("Fibonacci")
   print("0 1" )
   fibo(n-2)
   println()
   
   println("Fibonacci Recursivo")
   var x = 0;
   while( x < n ){
	 print( fib(x) )
	 x = x + 1
	 }

}